contracts/SimpleProviders/TimedDealProvider/TimedDealProvider.sol

```diff
@@ -49,23 +51,15 @@ contract TimedDealProvider is BasicProvider, TimedProviderState {
         return debitableAmount - (startAmount - leftAmount);
     }
 
-    function _calcRate(uint256 tokenAValue, uint256 tokenBValue) internal pure returns (uint256) {
-        return tokenBValue != 0 ? (tokenAValue * 1e18) / tokenBValue : 0;
-    }
-
-    function _calcAmount(uint256 amount, uint256 rate) internal pure returns (uint256) {
-        return rate != 0 ? amount * 1e18 / rate : 0;
-    }
-
     function split(
         uint256 oldPoolId,
         uint256 newPoolId,
         uint256 splitAmount
     ) public onlyProvider {
         uint256 leftAmount = lockDealProvider.dealProvider().getWithdrawableAmount(oldPoolId);
-        uint256 rate = _calcRate(leftAmount, splitAmount);
+        uint256 rate = CalcUtils.calcRate(leftAmount, splitAmount);
```

Comment:
By declaring the `using` keyword with a library, we can access its functions as if they were part of the current contract, without having to specify the library name each time. So we can update the code like this: 
```solidity
leftAmount.calcRate(splitAmount);
```
Or remove the `using` keyword and refer to `CalcUtils`.